src/Orleans.Runtime/Utilities/SiloUnobservedExceptionsHandler.cs

```diff
@@ -51,19 +51,17 @@ private void InternalUnobservedTaskExceptionHandler(object sender, UnobservedTas
         {
             var aggrException = e.Exception;
             var baseException = aggrException.GetBaseException();
-            var tplTask = (Task)sender;
-            var contextObj = tplTask.AsyncState;
-            var context = contextObj as ISchedulingContext;
+            var context = (sender as Task)?.AsyncState;
 
             try
             {
-                UnobservedExceptionHandler(context, baseException);
+                this.logger.LogError((int)ErrorCode.Runtime_Error_100104, baseException, "Silo caught an unobserved exception thrown from context {Context}: {Exception}", context, baseException);
```

Comment:
Missing $ for string interpolation?

Reply:
This is a logger template string, not interpolated - the variables Context & Exception will appear with their positional counterparts in logging systems (eg, columns in geneva or seq)

Reply:
Please do not put the exceptions into  the log message. The ILogger interface has a dedicated parameter for exceptions, there is no need to log them twice. I have issues with GCE with that because Google Stackdriver analyzes the logs and sometimes I get exceptions twice then.

Reply:
@SebastianStehle I'm happy to remove them in a follow up PR. The reason for doing this in the first place is that not all loggers emit the exception (eg, the Console logger does not - afaik)

Reply:
I think it does: https://github.com/dotnet/extensions/blob/master/src/Logging/Logging.Console/src/ConsoleLogger.cs#L135